sopel/modules/admin.py

```diff
@@ -247,61 +260,86 @@ def mode(bot, trigger):
+    :raises InvalidSectionOption: option does not exist for section
 
-    If there is no section, section will default to "core".
-    If value is None, the option will be deleted.
+    The ``value`` is optional and can be returned as ``None`` if omitted from command.
     """
-    # Get section and option from first argument.
     match = trigger.group(3)
     if match is None:
-        bot.reply("Usage: .set section.option value")
-        return
+        return False  # Invalid command
+
+    # Get section and option from first argument.
     arg1 = match.split('.')
     if len(arg1) == 1:
         section_name, option = "core", arg1[0]
     elif len(arg1) == 2:
         section_name, option = arg1
     else:
-        bot.reply("Usage: .set section.option value")
-        return
-    section = getattr(bot.config, section_name)
+        raise ValueError  # invalid command format
+
+    section = getattr(config, section_name, False)
+    if not section:
+        raise InvalidSection(section_name)
     static_sec = isinstance(section, StaticSection)
 
     if static_sec and not hasattr(section, option):
-        bot.say('[{}] section has no option {}.'.format(section_name, option))
-        return
+        raise InvalidSectionOption(section_name, option)  # Option not found in section
+
+    if not static_sec and not config.parser.has_option(section_name, option):
+        raise InvalidSectionOption(section_name, option)  # Option not found in section
 
     delim = trigger.group(2).find(' ')
     # Skip preceding whitespaces, if any.
     while delim > 0 and delim < len(trigger.group(2)) and trigger.group(2)[delim] == ' ':
         delim = delim + 1
 
-    # Display current value if no value is given.
+    value = trigger.group(2)[delim:]
     if delim == -1 or delim == len(trigger.group(2)):
-        if not static_sec and bot.config.parser.has_option(section, option):
-            bot.reply("Option %s.%s does not exist." % (section_name, option))
-            return
-        # Except if the option looks like a password. Censor those to stop them
-        # from being put on log files.
+        value = None
+
+    return (section, section_name, static_sec, option, value)
+
+
+@sopel.module.require_privmsg("This command only works as a private message.")
+@sopel.module.require_admin("This command requires admin privileges.")
+@sopel.module.commands('set')
+@sopel.module.example('.set core.owner Me')
+def set_config(bot, trigger):
+    """See and modify values of Sopel's config object.
+
+    Trigger args:
+        arg1 - section and option, in the form "section.option"
+        arg2 - value
+
+    If there is no section, section will default to "core".
+    If value is not provided, the current value will be displayed.
+    """
+    try:
+        section, section_name, static_sec, option, value = parse_section_option_value(bot.config, trigger)
+    except ValueError:
+        bot.reply('Usage: .set section.option [value]')
```

Comment:
Use `help_prefix` here instead of hardcoded `.`.

I know this and its counterpart are copypasta, but since you're touching this _anyway_ it makes sense to fix it.